From 4973579ab21161d429dc9992056a273b732167e7 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sun, 10 Aug 2014 18:19:54 -0400
Subject: [iommu] amd: Clean up unnecessary MSI/MSI-X capability find

Message-id: <20140810181954.24055.46972.stgit@gir.stowe>
Patchwork-id: 87575
O-Subject: [RHEL7.1 PATCH 04/19] iommu/amd: Clean up unnecessary MSI/MSI-X capability find
Bugzilla: 1127467
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Dutile <ddutile@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1127467
Upstream Status: 82fcfc674ebe2a0d83a6dd1b6ffa5589a94e4d4c

commit 82fcfc674ebe2a0d83a6dd1b6ffa5589a94e4d4c
Author: Yijing Wang <wangyijing@huawei.com>
Date:   Thu Aug 8 21:12:36 2013 +0800

    iommu/amd: Clean up unnecessary MSI/MSI-X capability find

    PCI core will initialize device MSI/MSI-X capability in
    pci_msi_init_pci_dev(). So device driver should use
    pci_dev->msi_cap/msix_cap to determine whether the device
    support MSI/MSI-X instead of using
    pci_find_capability(pci_dev, PCI_CAP_ID_MSI/MSIX).  Access
    to PCIe device config space again will consume more time.

    Signed-off-by: Yijing Wang <wangyijing@huawei.com>
    Cc: Joerg Roedel <joro@8bytes.org>
    Cc: iommu@lists.linux-foundation.org
    Cc: linux-kernel@vger.kernel.org
    Signed-off-by: Joerg Roedel <joro@8bytes.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/amd_iommu_init.c b/drivers/iommu/amd_iommu_init.c
index bf51abb..d9e92b9 100644
--- a/drivers/iommu/amd_iommu_init.c
+++ b/drivers/iommu/amd_iommu_init.c
@@ -1343,7 +1343,7 @@ static int iommu_init_msi(struct amd_iommu *iommu)
  if (iommu->int_enabled)
   goto enable_faults;
 
- if (pci_find_capability(iommu->dev, PCI_CAP_ID_MSI))
+ if (iommu->dev->msi_cap)
   ret = iommu_setup_msi(iommu);
  else
   ret = -ENODEV;
-- 
1.7.1